src/RevProc.cc

```diff
@@ -1564,6 +1574,31 @@ void RevProc::HandleALUFault(unsigned width){
                   "FAULT:ALU: ALU fault injected into next retire cycle\n");
 }
 
+bool RevProc::ThreadCanBeRemoved(const uint32_t& ThreadID) {
+  auto* regFile = AssignedThreads.at(ThreadID)->GetRegFile();
+
+  // Check the if regfile scoreboard is anything but clear 
+  // if not, this thread has outstanding requests and can't be removed
+  // TODO: (PRE-MERGE) Do this in a more ~Lee~ fashion
+  if( feature->IsRV32() ){
```

Comment:
`AnyDependency()` takes an `isFloat` parameter, but it can be changed to detect **any** dependency by ORing the RV and FP scoreboards, and then all of this can be replaced with `AnyDependency()`